CRM/Core/Menu.php

```diff
@@ -317,17 +317,21 @@ public static function store($truncate = TRUE) {
 
       $menu->find(TRUE);
 
-      // Move unrecognized fields to $module_data.
-      $module_data = array();
-      foreach (array_keys($item) as $key) {
-        if (!isset($daoFields[$key])) {
-          $module_data[$key] = $item[$key];
-          unset($item[$key]);
+      if (!CRM_Core_Config::isUpgradeMode() ||
+        CRM_Core_DAO::checkFieldExists('civicrm_menu', 'module_data', FALSE)
+      ) {
+        // Move unrecognized fields to $module_data.
+        $module_data = array();
+        foreach (array_keys($item) as $key) {
+          if (!isset($daoFields[$key])) {
+            $module_data[$key] = $item[$key];
+            unset($item[$key]);
+          }
         }
-      }
 
-      $menu->copyValues($item);
-      $menu->module_data = serialize($module_data);
+        $menu->copyValues($item);
```

Comment:
@highfalutin i think this should be moved down and outside the if the reason is we still need to load in any new data into the other db fields before saving. @totten thoughts?

Reply:
You're right. Should the `foreach` loop also be moved outside the `if`? This would unset unrecognized fields -- new behavior compared to previous releases.